### DISCUSSION TITLE
Inline text components

### DISCUSSION THREAD

**Discussion by amitu** [fastn-stack/fastn#1972] [Ideas & RFCs]
```
# this uses #foo[inline styled] argument

- it can be anywhere #foo[whatever]


-- styled-text foo:
;; this is a special component that only gets `caption` as the component argument

-- ftd.text: $foo.caption
color: red

-- end: foo
```

Not too fond of this as it can not be used for something as simple as a link or an inline image, as they need more than one bits of input. 

Not yet sure if I like this over https://github.com/orgs/fastn-stack/discussions/1954 tho.

**Comment by amitu** [comment]
`B<text to show as bold>` `B` is a component. `-- foo: hello, ds.bold<amit>?`

**Comment by siddhantk232** [comment]
I too agree with #1954 more than this one.
